x-pack/test/apm_api_integration/tests/mobile/mobile_location_stats.spec.ts

```diff
@@ -347,6 +347,28 @@ export default function ApiTest({ getService }: FtrProviderContext) {
           kuery: `service.version:"1.1" and service.environment: "production"`,
         });
 
+        const timestamps = dataGenerated.timestamps;
+        expect(
+          response.currentPeriod.mostSessions.timeseries.every((item) =>
+            timestamps.includes(item.x) ? item.y === 1 : item.y === 0
```

Comment:
Not sure if that test will prevent https://github.com/elastic/kibana/pull/168925/commits/1f794c9e35ef5620cb8d56146e20bc600ee92194#diff-fb1548053a17e885dd4f5878c8c8db81aa19ad0b376c25d70e13c07696fa4ebcR104 from happening again. 

In that case, I think it's better to explicitly test the time series for example . 

The first  value
```
const firstValue = response.currentPeriod.mostSessions.timeseries[0].y
expect(firstValue).to.be('Y')
```
The last value

```
import { last } from 'lodash';

const lastValue = last(response.currentPeriod.mostSessions.timeseries)?.y
expect(lastValue).to.be('Y')
```

or by checking if all the `y` values are different.

Reply:
Sorry I'm not sure I understand properly. Let me provide some context on the behavior I noticed that led me to create the test this way to see if it makes sense:

- There are 3 transactions generated [here](https://github.com/elastic/kibana/pull/168925/files#diff-1f9f849aa986818c0bbc4123f45fe2384f4df32fe3721ebf6c7ef2526cb1c661R148).
- The number of items inside `response.currentPeriod.mostSessions.timeseries` were something like 11 iirc out of which only 3 had their `Y` value set to `1` and the rest of them was set to `0`.
- The `X` values whose `Y` was set to 1 were the timestamps for the 3 transactions generated  [here](https://github.com/elastic/kibana/pull/168925/files#diff-1f9f849aa986818c0bbc4123f45fe2384f4df32fe3721ebf6c7ef2526cb1c661R148), so I gathered those timestamps during the data generation [here](https://github.com/elastic/kibana/pull/168925/files#diff-1f9f849aa986818c0bbc4123f45fe2384f4df32fe3721ebf6c7ef2526cb1c661R144) in order to later check that those are the only ones which represent `Y` set to 1, and the rest should be `0`.

> I think it's better to explicitly test the time series for example

You mean to only validate the values that should contain `y = 1`?

> or by checking if all the y values are different.

All the items that don't represent the 3 generated transactions have their `y` value set to `0`, so if I understood correctly, this test could fail since those won't be different, similar to the 3 values set to `1` 😅  Though maybe I didn't get it properly.

Before making [this change](https://github.com/elastic/kibana/commit/1f794c9e35ef5620cb8d56146e20bc600ee92194#diff-fb1548053a17e885dd4f5878c8c8db81aa19ad0b376c25d70e13c07696fa4ebcR104), I noticed that this test was failing because all the `Y` values of the 11 items were set to `3`.